pkg/workflow/management/upgrade.go

```diff
@@ -0,0 +1,41 @@
+package management
+
+import "github.com/aws/eks-anywhere/pkg/workflow"
+
+const (
+	PreUpgradeClusterTaskName  workflow.TaskName = "PreUpgradeManagementCluster"
+	PostUpgradeClusterTaskName workflow.TaskName = "PostUpgradeManagementCluster"
+)
+
+// UpgradeClusterHookRegistrar is a Hook registrar that binds hooks to an upgrade management
+// cluster workflow.
+type UpgradeClusterHookRegistrar interface {
+	RegisterUpgradeManagementClusterHooks(workflow.HookBinder)
+}
+
+// UpgradeClusterBuilder defines the configuration for a management cluster upgrade workflow.
+type UpgradeClusterBuilder struct {
+	HookRegistrars []UpgradeClusterHookRegistrar
+}
+
+// WithHookRegistrar adds a hook registrar to the upgrade cluster workflow builder.
+func (b *UpgradeClusterBuilder) WithHookRegistrar(registrar UpgradeClusterHookRegistrar) *UpgradeClusterBuilder {
```

Comment:
If we were gonna have this kind of method (see my comment above), this would have my vote. It does require to re-implement this method for every workflow builder, but given it's two lines and we don't expect the number of workflows to wrong significantly, it a good tradeoff for the cognitive complexity it removes.

That said, if y'all agree the generics approach is better, it wouldn't concern me much.

Reply:
I've updated everything with this approach. Its easy enough to return to a generic approach if we want.